extensions/wikia/Discussions/maintenance/ForumDumper.php

```diff
@@ -0,0 +1,269 @@
+			} );
+
+		return $this->pages;
+	}
+
+	public function getRevisions() {
+		if ( !empty( $this->revisions ) ) {
+			return $this->revisions;
+		}
+		
+		$pageIds = array_keys( $this->getPages() );
+
+		$dbh = wfGetDB( DB_SLAVE );
+		( new \WikiaSQL() )
+			->SELECT_ALL()
+			->FROM( self::TABLE_REVISION )
+			->JOIN( self::TABLE_TEXT )->ON( 'rev_text_id', 'old_id' )
+			->WHERE( 'rev_page' )->IN( $pageIds )
+			->runLoop( $dbh, function ( &$revisions, $row ) {
+				$textId = $row->old_text;
+				list( $parsedText, $plainText, $title ) = $this->getTextAndTitle( $textId );
+
+				$pages = $this->getPages();
+				$curPage = $pages[$row->rev_page];
+				
+				$this->addRevision( [
+					"revision_id" => $row->rev_id,
+					"page_id" => $row->rev_page,
+					"page_namespace" => $curPage['namespace'],
+					"title" => $title,
+					"user_type" => $this->getContributorType( $row ),
+					"user_identifier" => $row->rev_user,
+					"timestamp" => $row->rev_timestamp,
+					"is_minor_edit" => $row->rev_minor_edit,
+					"is_deleted" => $row->rev_deleted,
+					"length" => $row->rev_len,
+					"parent_id" => $row->rev_parent_id,
+					"text_flags" => $row->old_flags,
+					"comment" => $row->rev_comment,
+					"raw_content" => $plainText,
+				    "content" => $parsedText
+				] );
+			} );
+
+		return $this->revisions;
+	}
+
+	public function getTextAndTitle( $textId ) {
+		$rawText = $this->getRawText( $textId );
+
+		// The title is included within the text as an ac_metadata tag
+		$title = $this->getTitle( $rawText );
+
+		// Parse the wiki text
+		$parsedText = $this->getParsedText( $rawText );
+
+		// Get a plain text version as well
+		$plainText = strip_tags( $parsedText );
+
+		return [ $parsedText, $plainText, $title ];
+	}
+
+	private function getRawText( $textId ) {
+		$text = \ExternalStore::fetchFromURL( $textId );
+		return gzinflate( $text );
+	}
+
+	private function getParsedText( $wikiText ) {
+		// If this text appears not to have any markup, just return the text as is.
+		if ( !preg_match( self::REGEXP_MATCH_HAS_MARKUP, $wikiText ) ) {
+			return $wikiText;
+		}
+
+		$parserOut = \F::app()->wg->Parser->parse(
+			$wikiText,
+			$this->dummyTitle,
+			$this->parserOptions
+		);
+		return $parserOut->getText();
+	}
+
+	private function getTitle( $text ) {
+		$title = '';
+
+		if ( preg_match( self::REGEXP_MATCH_TITLE, $text, $matches ) ) {
+			$title = $matches[1];
+		}
+
+		return $title;
+	}
+
+	public function getContributorType( $row ) {
+		if ( empty( $row->rev_user_text ) ) {
+			return self::CONTRIBUTOR_TYPE_UNKNOWN;
+		}
+
+		if ( $row->rev_deleted & \Revision::DELETED_USER ) {
+			return self::CONTRIBUTOR_TYPE_DELETED;
+		}
+
+		if ( \IP::isIPv4( $row->rev_user_text ) ) {
+			return self::CONTRIBUTOR_TYPE_IP;
+		}
+
+		return self::CONTRIBUTOR_TYPE_USER;
+	}
+
+	public function getVotes() {
+		if ( !empty( $this->votes ) ) {
+			return $this->votes;
+		}
+
+		$pageIds = array_keys( $this->getPages() );
+
+		$dumper = $this;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$dumper" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)